Cesium.CodeGen/Ir/Expressions/CompoundObjectInitializationExpression.cs

```diff
@@ -0,0 +1,260 @@
+using Cesium.Ast;
+using Cesium.CodeGen.Contexts;
+using Cesium.CodeGen.Extensions;
+using Cesium.CodeGen.Ir.Declarations;
+using Cesium.CodeGen.Ir.Types;
+using Cesium.Core;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+using System.Collections.Immutable;
+using System.Xml.Linq;
+
+namespace Cesium.CodeGen.Ir.Expressions;
+internal sealed class CompoundObjectInitializationExpression : IExpression
+{
+    private IType? _type;
+    private TypeDefinition? _typeDef;
+    private readonly ImmutableArray<IExpression?> _initializers;
+    private IDeclarationScope? _scope;
+
+    public CompoundObjectInitializationExpression(IType type, ImmutableArray<IExpression?> initializers)
+    {
+        _type = type;
+        _initializers = initializers;
+    }
+
+    public CompoundObjectInitializationExpression(ImmutableArray<IExpression?> initializers)
+    {
+        _initializers = initializers;
+    }
+
+    public void Hint(TypeDefinition type) => _typeDef = type;
+
+    public void EmitTo(IEmitScope scope)
+    {
+        if (_type == null && _typeDef == null)
+            throw new Exception("_type is null!");
+
+        var instructions = scope.Method.Body.Instructions;
+        TypeDefinition typeDef = _type != null ? ((StructType)_type).Resolve(scope.Context).Resolve() : _typeDef!;
+        var fieldsDefs = typeDef.Fields;
+        var initializers = _initializers;
+
+        if (typeDef.Name.Contains("<SyntheticBuffer>"))
```

Comment:
I'd suggest to introduce some helper method that would do this check for us. I am not comfortable with the idea of random pieces of code just performing type classification based on type names.

Let's add it to `TypeSystemEx` perhaps?

Reply:
Like: `TypeSystemEx.IsArray(TypeRefernce typeRef) => typeRef.Name.StartsWith("<SyntheticBuffer>");` ?

Reply:
Yep, exactly like this.